meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -315,7 +315,7 @@ public ConnectState getClientConnectionState() {
   @Override
   public boolean subscribeDataChange(String key, DataChangeListener listener, boolean skipWatchingNonExistNode) {
     _zkClient.subscribeDataChanges(key, new DataListenerAdapter(listener));
-    return false;
+    return true;
```

Comment:
default "true" return type means that this method should be void.

Reply:
Good point. Will do interface change in following PR.